src/pudl/transform/eia861.py

```diff
@@ -2327,76 +2146,290 @@ def utility_data(tfr_dfs):
+        "demand_side_management_sales_eia861": AssetIn(),
+        "distributed_generation_fuel_eia861": AssetIn(),
+        "distributed_generation_misc_eia861": AssetIn(),
+        "distributed_generation_tech_eia861": AssetIn(),
+        "distribution_systems_eia861": AssetIn(),
+        "dynamic_pricing_eia861": AssetIn(),
+        "energy_efficiency_eia861": AssetIn(),
+        "green_pricing_eia861": AssetIn(),
+        "mergers_eia861": AssetIn(),
+        "net_metering_customer_fuel_class_eia861": AssetIn(),
+        "net_metering_misc_eia861": AssetIn(),
+        "non_net_metering_customer_fuel_class_eia861": AssetIn(),
+        "non_net_metering_misc_eia861": AssetIn(),
+        "operational_data_misc_eia861": AssetIn(),
+        "operational_data_revenue_eia861": AssetIn(),
+        "reliability_eia861": AssetIn(),
+        "sales_eia861": AssetIn(),
+        "utility_data_misc_eia861": AssetIn(),
+        "utility_data_nerc_eia861": AssetIn(),
+        "utility_data_rto_eia861": AssetIn(),
+    },
+)
+def utility_assn_eia861(**data_dfs: dict[str, pd.DataFrame]) -> pd.DataFrame:
+    """Harvest a Utility-Date-State Association Table."""
+    logger.info("Building an EIA 861 Util-State-Date association table.")
+    df = _harvest_associations(
+        dfs=list(data_dfs.values()), cols=["report_date", "utility_id_eia", "state"]
+    ).pipe(_post_process)
+    return df
 
 
-def transform(raw_dfs, eia861_settings: Eia861Settings = Eia861Settings()):
-    """Transform EIA 861 DataFrames.
+@asset(
+    ins={
+        "advanced_metering_infrastructure_eia861": AssetIn(),
+        "clean_balancing_authority_eia861": AssetIn(),
+        "demand_response_eia861": AssetIn(),
+        "demand_response_water_heater_eia861": AssetIn(),
+        "demand_side_management_ee_dr_eia861": AssetIn(),
+        "demand_side_management_misc_eia861": AssetIn(),
+        "demand_side_management_sales_eia861": AssetIn(),
+        "distributed_generation_fuel_eia861": AssetIn(),
+        "distributed_generation_misc_eia861": AssetIn(),
+        "distributed_generation_tech_eia861": AssetIn(),
+        "distribution_systems_eia861": AssetIn(),
+        "dynamic_pricing_eia861": AssetIn(),
+        "energy_efficiency_eia861": AssetIn(),
+        "green_pricing_eia861": AssetIn(),
+        "mergers_eia861": AssetIn(),
+        "net_metering_customer_fuel_class_eia861": AssetIn(),
+        "net_metering_misc_eia861": AssetIn(),
+        "non_net_metering_customer_fuel_class_eia861": AssetIn(),
+        "non_net_metering_misc_eia861": AssetIn(),
+        "operational_data_misc_eia861": AssetIn(),
+        "operational_data_revenue_eia861": AssetIn(),
+        "reliability_eia861": AssetIn(),
+        "sales_eia861": AssetIn(),
+        "utility_data_misc_eia861": AssetIn(),
+        "utility_data_nerc_eia861": AssetIn(),
+        "utility_data_rto_eia861": AssetIn(),
+    },
+)
+def balancing_authority_assn_eia861(**dfs: dict[str, pd.DataFrame]) -> pd.DataFrame:
```

Comment:
Moved this function down from above, to group it with the other "secondary transform" functions.